packages/schema-blocks/src/instructions/blocks/VariationPicker.tsx

```diff
@@ -6,11 +6,22 @@ import BlockLeaf from "../../core/blocks/BlockLeaf";
 // @ts-ignore -- __experimentalBlockVariationPicker is defined in the package, though no type info is available.
 import { __experimentalBlockVariationPicker as ExperimentalBlockVariationPicker, useBlockProps } from "@wordpress/block-editor";
 import { get, map } from "lodash";
-import { createBlock } from "@wordpress/blocks";
+import { BlockInstance, createBlock } from "@wordpress/blocks";
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 import { createElement } from "@wordpress/element";
-import { BlockInstance } from "@wordpress/blocks";
 import { VariationInterface } from "./Variation";
+import { BlockValidationResult } from "../../core/validation";
+
+/**
+ * Helper function to check whether the block instance includes a picked variation.
+ *
+ * @param blockInstance The block instance to check.
+ *
+ * @returns Whether the block instance includes a variation.
+ */
+function includesAVariation( blockInstance: BlockInstance ): boolean {
+	return blockInstance.innerBlocks && blockInstance.innerBlocks.length > 0;
```

Comment:
how is having innerblocks a sign of having variations? at least one of the innerblocks must implement the VariationImplementation interface. (not entirely sure about that name)

```suggestion
	return blockInstance.innerBlocks && blockinstance.innerBlocks.some( block => block instanceof VariationImplementation );
```

Reply:
I can't seem to find a way to tell whether an inner block has been added as a variation or not 😕 .

When a variation has been picked, the inner blocks are replaced with the blocks in the variation.